prereise/gather/hiflddata/plant_agg.py

```diff
@@ -0,0 +1,387 @@
+    return region
+
+
+def Plant_agg(
+    clean_data, ZipOfsub_dict, loc_of_plant, LocOfsub_dict, Pmin, region, points
+):
+    """Aggregate the plant by zip code and build the plant dict with geo-based aggregation
+
+    :return: (*dict*) -- a dict for power plant after aggregation
+    """
+
+    plant_dict = {}
+    sto = [
+        "HYDROELECTRIC PUMPED STORAGE",
+        "BATTERIES",
+        "NATURAL GAS WITH COMPRESSED AIR STORAGE",
+        "SOLAR THERMAL WITH ENERGY STORAGE",
+        "FLYWHEELS",
+    ]
+
+    for index, row in clean_data.iterrows():
+        tu = (row["PLANT"], row["PRIM_FUEL"], row["PRIM_MVR"])
+        u = (row["PLANT"], row["PRIM_FUEL"])
+        r = (row["PLANT"], row["NAME"])
+        if row["TYPE"] in sto:
+            continue
+        if tu not in plant_dict:
+            if u in Pmin and row["PLANT"] in loc_of_plant:
+                if row["ZIP"] in ZipOfsub_dict:
+                    min_d = 100000.0
+                    for value in ZipOfsub_dict[row["ZIP"]]:
+                        # calculate the distance between the plant and substations
+                        if (
+                            haversine(loc_of_plant[row["PLANT"]], LocOfsub_dict[value])
+                            < min_d
+                        ):
+                            min_d = haversine(
+                                loc_of_plant[row["PLANT"]], LocOfsub_dict[value]
+                            )
+                    bus_id = value
+                    pmin = Pmin[u]
+                # if this zip does not contain subs, we try to find subs in neighbor zip.
+                else:
+                    zi = int(row["ZIP"])
+                    for i in range(-5, 6):
+                        min_d = 100000.0
+                        if str(zi + i) in Pmin and str(zi + i) in loc_of_plant:
+                            for value in ZipOfsub_dict[str(zi + i)]:
+                                if (
+                                    haversine(
+                                        loc_of_plant[row["PLANT"]], LocOfsub_dict[value]
+                                    )
+                                    < min_d
+                                ):
+                                    min_d = haversine(
+                                        loc_of_plant[row["PLANT"]], LocOfsub_dict[value]
+                                    )
+                    bus_id = value
+                    pmin = 0
+            pmaxwin = row["WINTER_CAP"]
+            if pmaxwin < 0.0:
+                pmaxwin = 0.0
+            pmaxsum = row["SUMMER_CAP"]
+            if pmaxsum < 0.0:
+                pmaxsum = 0.0
+            if r in points:
+                cur = points[r]
+
+            else:
+                cur = 0
+            if r in region:
+                re = region[r]
+            else:
+                if row["STATE"] in West:
+                    re = "Western"
+                elif row["STATE"] is "TX":
```

Comment:
When running this script, I get a warning at this line:
```python
plant_agg.py:200: SyntaxWarning: "is" with a literal. Did you mean "=="?
  elif row["STATE"] is "TX":
```
I believe `is` and `==` have different meaning here.